conan/tools/scons/sconsdeps.py

```diff
@@ -0,0 +1,62 @@
+from jinja2 import Template
+from conan.tools._check_build_profile import check_using_build_profile
+from conans.model.new_build_info import NewCppInfo
+from conans.util.files import save
+
+
+class SConsDeps:
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._ordered_deps = []
+        self._generator_file = 'SConscript_conandeps'
+        check_using_build_profile(self._conanfile)
+
+    @property
+    def ordered_deps(self):
+        if not self._ordered_deps:
+            deps = self._conanfile.dependencies.host.topological_sort
+            self._ordered_deps = [dep for dep in reversed(deps.values())]
+        return self._ordered_deps
+
+    def _get_cpp_info(self):
+        ret = NewCppInfo()
+        for dep in self.ordered_deps:
+            dep_cppinfo = dep.cpp_info.aggregated_components()
+            ret.merge(dep_cppinfo)
+        return ret
+
+    def generate(self):
+        save(self._generator_file, self._content)
+
+    @property
+    def _content(self):
+        template = Template("""
+        "{{dep_name}}" : {
+            "CPPPATH"     : {{info.includedirs or []}},
+            "LIBPATH"     : {{info.libdirs or []}},
+            "BINPATH"     : {{info.bindirs or []}},
+            "LIBS"        : {{(info.libs or []) + (info.system_libs or [])}},
+            "FRAMEWORKS"  : {{info.frameworks or []}},
+            "FRAMEWORKPATH" : {{info.frameworkdirs or []}},
+            "CPPDEFINES"  : {{info.defines or []}},
+            "CXXFLAGS"    : {{info.cxxflags or []}},
+            "CCFLAGS"     : {{info.cflags or []}},
+            "SHLINKFLAGS" : {{info.sharedlinkflags or []}},
+            "LINKFLAGS"   : {{info.exelinkflags or []}},
+        },
+        {% if dep_version is not none %}"{{dep_name}}_version" : "{{dep_version}}",{% endif %}
+        """)
+        sections = ["conan = {\n"]
```

Comment:
Wasn't the section called ``conandeps``? (to account for a potential ``conantoolchain``)

Reply:
Right now, as it is is this:

```
# SConscript_conandeps generated by conan
conan = {
        # this one aggregates all libs
        "conandeps" : {
            "CPPPATH" ...
        },
        
        # now separate libs
        "zlib" : {
            "CPPPATH" ...
        },
        "zlib_version" ...
        ...
```

loaded like this in the SConscript:

```
# SConscript by the consumer
conan = SConscript('./SConscript_conandeps')
flags = conan["conandeps"]
env.MergeFlags(flags)
```

Maybe we can do:

```
conandeps = {
        # this one aggregates all libs
        "conanlibs" : { # all_libs? aggregated_libs?...
            "CPPPATH" ...
        },
        ...
```

then:

```
conan = SConscript('./SConscript_conandeps')
flags = conandeps["conanlibs"]
env.MergeFlags(flags)
```

Something like that?